libs/community/tests/integration_tests/vectorstores/test_aperturedb.py

```diff
@@ -0,0 +1,203 @@
+"""Test ApertureDB functionality."""
+
```

Comment:
Could you test this with our standard test suite? if it passes the unit tests we can merge.

https://github.com/langchain-ai/langchain/blob/14a8bbc21ac81f4a0eb03db981780c86cbdb9a7f/libs/community/tests/unit_tests/vectorstores/test_inmemory.py#L6-L6

Reply:
Ah, I didn't see those.  I was going by existing examples.  I'll convert to that.

Reply:
It was recently added, we're in the process of adopting this in more places -- it'll make it easier for vectorstore maintainers to verify that their implementation is correct and hopefully be able to avoid re-inventing all the test scaffolding

Reply:
In `test_deleting_documents`, I see:
```
ids = vectorstore.add_documents(documents, ids=["1", "2"])
```
Is it a new convention that `add_documents` should support an `ids` parameter?  Shouldn't we be using the `id` field in Document instead?

Reply:
Not a new convention -- unfortunately it's an OLD API and most implementations are expected to support it. If you implement just upsert  and drop add_documents, the correct behavior will be implemented for you including correct precedence rules for `ids` provided via add_documents vs. ids on the document object itself

Reply:
OK.  I have converted to the standard tests and they run successfully.  

I had a bit of a journey to getting the devcontainer working. 

First I had to change the Poetry version by setting `POETRY_VERSION=1.4.2` in `dev.Dockerfile`

Then I had to execute the following commands:
```bash
pip3 install pytest-vcr
pip3 install -e libs/standard-tests/
pip3 install aperturedb
adb config create host.docker.internal --host host.docker.internal --no-interactive --active
```

I also had to add the following In `.devcontainer/docker-compose.yaml` to `services/langchain`:
```yaml
extra_hosts:
    "host.docker.internal:host-gateway" 
```

Reply:
most of the core developers don't use the devcontainer as far as I know -- I think it's mostly maintained by the community. I'm happy to merge the changes for the dev container

Reply:
> I'm happy to merge the changes for the dev container

I'm planning to submit a separate PR for that.